src/randoop/sequence/AbstractGenerator.java

```diff
@@ -155,34 +168,74 @@ public AbstractGenerator(List<Operation> operations, long timeMillis, int maxSeq
     }
     
     this.stopper = stopper;
-    
     this.listenerMgr = listenerManager;
-    
-    outputTestFilters = new LinkedList<ITestFilter>();
-    if (testfilters == null || testfilters.isEmpty()) {
-      outputTestFilters.add(new DefaultTestFilter());
-    } else {
-      outputTestFilters.addAll(testfilters);
+  }
+
+  /**
+   * Registers test predicate with the generator object for use while filtering
+   * generated tests.
+   * 
+   * @param outputTest  the predicate to be added to object
+   */
+  public void addTestPredicate(Predicate<ExecutableSequence> outputTest) {
+    if (outputTest == null) {
+      throw new IllegalArgumentException("outputTest must be non-null");
     }
+    this.outputTest = outputTest;
   }
 
+  /**
+   * Registers a visitor with this object for use while executing generated
+   * sequences.
+   * 
+   * @param executionVisitor  the visitor
+   */
+  public void addExecutionVisitor(ExecutionVisitor executionVisitor) {
+    if (executionVisitor == null) {
+      throw new IllegalArgumentException("executionVisitor must be non-null");
+    }
+    this.executionVisitor = executionVisitor;
+  }
+  
+  /**
+   * Registers a visitor with this object to generate checks following
```

Comment:
I initially read this as "run the visitor after all generated tests have been executed".  But I think you mean that it is run after each generated test is executed.  Please clarify here and for other nearby methods' documentation.